cmake/helpers/CheckDependentLibraries.cmake

```diff
@@ -597,6 +597,15 @@ if (ARMADILLO_FOUND)
     set(GDAL_USE_ARMADILLO CACHE BOOL OFF FORCE)
   endif ()
 
+  # LAPACK support required for arma::solve()
+  if (GDAL_USE_ARMADILLO AND EXISTS "${ARMADILLO_INCLUDE_DIRS}/armadillo_bits/config.hpp")
+      file(READ "${ARMADILLO_INCLUDE_DIRS}/armadillo_bits/config.hpp" armadillo_config)
+      if ("${armadillo_config}" MATCHES "/\\* #undef ARMA_USE_LAPACK")
+          message(WARNING "Armadillo build lacks LAPACK support. Disabling it as it cannot be used by GDAL")
```

Comment:
"lacks" or "uses" ?  The PR title and commit message seem to be wrong.

Reply:
I also wonder it shouldn't be a fatal error if the user explicitly enabled Armadillo.

Reply:
> The PR title and commit message seem to be wrong.

indeed. changed "CMake: do not use Armadillo if it *lacks* LAPACK support (such as on Alpine)". Thanks!

Reply:
> I also wonder it shouldn't be a fatal error if the user explicitly enabled Armadillo.

good idea. done